src/shogun/features/RandomFourierDotFeatures.h

```diff
@@ -103,7 +103,13 @@ class RandomFourierDotFeatures : public RandomKitchenSinksDotFeatures
 	 *
 	 * @return a random parameter vector
 	 */
-	virtual SGVector<float64_t> generate_random_parameter_vector();
+	virtual void generate_random_parameter_vector(SGVector<float64_t>&& vec);
+
+	/**  Get number of parameters per sample in the random_coeff matrix
```

Comment:
dont reference variables in comments.
Just "@return number of parameters per input dimension"